### PR TITLE
HAAR-862: end point to retrieve service details by service code

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesControllerTest.kt [hunk 1]
```diff
@@ -10,30 +12,59 @@ import org.springframework.security.core.Authentication
...
+
     val response = controller.services()
-    assertThat(response).containsExactly(AuthService(auth1), AuthService(auth2))
+
+    assertThat(response).containsExactly(AuthService(services[0]), AuthService(services[1]))
   }
 
   @Test
   fun `test services returns all my services`() {
-    val auth1 = Service("code1", "name1", "description1", "roles1", "url1", true, "email1")
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesControllerTest.kt:30]: Do we need to test returning no services

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesController.kt [hunk 2]
```diff
@@ -6,6 +6,7 @@ import io.swagger.v3.oas.annotations.media.Schema.AccessMode.READ_ONLY
 import io.swagger.v3.oas.annotations.tags.Tag
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesController.kt:6]: Should probably add e2e tests in AuthServicesIntTest

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesControllerTest.kt (hunk 1 above)
Comment (AlTurner-MOJ) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesControllerTest.kt:30]: That was an existing test case for an existing end point - I altered it only to make the overall code in the test more succinct. Agree that test case appears to be missing, but feels like adding it under this ticket would be bit off-piste?

Comment (AlTurner-MOJ) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesControllerTest.kt:30]: The end point I added either returns a single Service or NOT FOUND

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesController.kt (hunk 2 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesController.kt:6]: Adding - thought there were no existing integration tests

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesIntTest.kt [hunk 3]
```diff
@@ -71,4 +72,39 @@ class AuthServicesIntTest : IntegrationTest() {
...
+    }
+
+    @Test
+    fun `Responds with service details when exists`() {
+      webTestClient.get().uri("/api/services/book-a-secure-move-ui")
+        .headers(setAuthorisation("AUTH_ADM", listOf("ROLE_MAINTAIN_OAUTH_USERS")))
+        .exchange()
+        .expectStatus().isOk
+    }
+  }
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesIntTest.kt:109]: Just need to check header/body/jsondata is correct

Comment (AlTurner-MOJ) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/AuthServicesIntTest.kt]: Done

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**